wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wom/graph/CallElementToGraphNode.scala

```diff
@@ -2,34 +2,126 @@ package wdl.draft3.transforms.wdlom2wom.graph
 
 import cats.syntax.validated._
 import cats.syntax.either._
-import common.validation.ErrorOr.ErrorOr
-import wdl.model.draft3.elements.{CallElement, IfElement}
+import cats.instances.list._
+import cats.syntax.foldable._
+import cats.instances.option._
+import cats.instances.vector._
+import cats.syntax.validated._
+import cats.instances.map._
+import cats.syntax.traverse._
+import common.validation.ErrorOr._
+import shapeless.Coproduct
+import wdl.draft3.transforms.wdlom2wom.expression.WdlomWomExpression
+import wdl.model.draft3.elements.CallElement
 import wdl.model.draft3.graph.{GeneratedValueHandle, UnlinkedConsumedValueHook}
-import wom.callable.{Callable, CallableTaskDefinition, CommandTaskDefinition, TaskDefinition}
-import wom.graph.CallNode.{CallNodeBuilder, InputDefinitionFold}
-import wom.{callable, graph}
+import wom.callable.Callable.{InputDefinition, InputDefinitionWithDefault, OptionalInputDefinition, RequiredInputDefinition}
+import wom.callable.Callable
+import wom.graph.CallNode.{CallNodeBuilder, InputDefinitionFold, InputDefinitionPointer}
+import wom.graph.GraphNode.GraphNodeSetter
 import wom.graph.GraphNodePort.OutputPort
-import wom.graph.{CommandCallNode, GraphNode, WomIdentifier}
+import wom.graph.expression.{AnonymousExpressionNode, ExpressionNode, PlainAnonymousExpressionNode, TaskCallInputExpressionNode}
+import wom.graph._
 import wom.types.WomType
 
-object CallElementToGraphNode extends Object {
-  def convert(a: CallableNodeMakerInputs): ErrorOr[Set[GraphNode]] = {
-    val callable: ErrorOr[Callable] = a.callables.find(_.name == a.node.callableName) match {
-      case Some(task: CommandTaskDefinition) => task.valid
-      case Some(c: Callable) => c.valid
-      case None => s"Cannot resolve a callable with name ${a.node.callableName}".invalidNel
+object CallElementToGraphNode {
+  def convert(a: CallNodeMakerInputs): ErrorOr[Set[GraphNode]] = {
+    val callNodeBuilder = new CallNode.CallNodeBuilder()
+
+    /**
+      * Each input definition KV pair becomes an entry, LocalName(key) mapped to ExpressionNode(value).
+      *
+      * i.e.
+      * call foo {
+      *   input: key = value
+      * }
+      */
+    def expressionNodeMappings: ErrorOr[Map[LocalName, AnonymousExpressionNode]] = {
+      a.node.body match {
+        case Some(body) =>
+          body.inputs.map(input => input.key -> input.value).toMap.traverse {
+            case (name, expression) =>
+              val identifier = WomIdentifier(name)
+              val constructor = a.callableValidation match {
+                case _ => PlainAnonymousExpressionNode.apply _
```

Comment:
It looks like the draft2 version was using `TaskCallInputExpressionNode.apply _` if it was a Task and the `PlainAnonymousExpressionNode.apply _` in other cases (workflows)...

@Horneth do you know what that distinction is all about? We should probably do the same thing in any case